### PR TITLE
Fix/trailing slash

### PR DESCRIPTION
**Pull request by phixid**
We fixed an issue where passing a Cerberus url that ends in a slash is sometimes breaking the retrieval of secrets.

- fixed a typo
- renamed existing class method for clarity and unification
- added class method for handling trailing slashes
- used new class method in client initialization

### CODE REVIEW

File: cerberus/client.py [hunk 1]
```diff
@@ -69,12 +69,19 @@ def __init__(self, cerberus_url, username=None, password=None,
...
     # noinspection PyMethodMayBeStatic
-    def _add_slash(self, string=None):
+    def _add_trailing_slash(self, string=None):
         """ if a string doesn't end in a '/' add one """
         if not str.endswith(string, '/'):
             return str.join('', [string, '/'])
         return str(string)
 
+    # noinspection PyMethodMayBeStatic
+    def _remove_trailing_slash(self, string=None):
```
Comment (mayitbeegh) [inline cerberus/client.py:79]: I know that the string=None part is borrowed from the `_add_trailing_slash` function but I don't know if having a default value makes sense.

Comment (mayitbeegh) [reply cerberus/client.py:79]: Also can you add some unit tests in test_cerberus_client.py?

Comment (ToneVDB) [reply cerberus/client.py]: We made some changes, please re-review them :)

File: tests/test_url_util.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
+"""
+Copyright 2018-present Nike, Inc.
```
Comment (mayitbeegh) [inline tests/test_url_util.py:2]: Just nitpicking, this should be 2020

File: tests/test_cerberus_client.py [hunk 3]
```diff
@@ -453,30 +455,37 @@ def test_list_files(self, mock_get):
...
+            'secure_file_summaries': [
+                {
+                    'sdbox_id': '244cfc0d-4beb-8189-5056-194f18ead6f4',
+                    'path': 'studio54/test.py',
+                    'size_in_bytes': 1323,
+                    'name': 'test.py',
+                    'created_by': 'tester@studio54.com',
+                    'created_ts': '1974-11-17T00:02:30Z',
+                    'last_updated_by': 'tester@studio54.com',
+                    'last_updated_ts': '1974-11-17T00:02:30Z'},
```
Comment (mayitbeegh) [inline tests/test_cerberus_client.py:467]: Just nitpicking, the closing bracket should probably go into a new line

File: tests/test_url_util.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        assert url_with_trailing_slash.endswith(".com/")
+
+    @staticmethod
+    def test_multiple_trailing_slash():
+        url_with_trailing_slash = ensure_single_trailing_slash("nike.com//")
+        assert url_with_trailing_slash.endswith(".com/")
+
+
+class TestEnsureNoTrailingSlash(unittest.TestCase):
+    """unit tests for url_util.ensure_trailing_slash"""
```
Comment (mayitbeegh) [inline tests/test_url_util.py:40]: should  be `ensure_no_trailing_slash`

File: cerberus/client.py (hunk 1 above)
Comment (mayitbeegh) [reply cerberus/client.py]: I added some comments for minor changes but otherwise it looks good! And thanks for reformatting.

**Comment by phixid** [comment]
Thank you very much for the thorough review 👍. Hope I fixed them all now.

**Comment by ToneVDB** [comment]
Any other steps we should take to get this released?

**Comment by mayitbeegh** [comment]
> Any other steps we should take to get this released?

Sorry I didn't see the update, everything looks good to go!

**Pull request merged**